### ISSUE TITLE
_DownloadedDataset does not cache the transformed dataset

### ISSUE THREAD

**Issue by marfago** [apache/incubator-mxnet#8909]
In the current implementation of the python class _DownloadedDataset the __getitem__ applies always the transformation to the raw data causing an extra load in calculation.

This cause the training/test operations in the examples to be much slower than needed (#8708).

A simple workaround is to rewrite this code
```
def transform(data, label):
    return data.astype(np.float32)/255, label.astype(np.float32)
train_data = mx.gluon.data.DataLoader(mx.gluon.data.vision.MNIST(train=True, transform=transform),
                                      batch_size, shuffle=True)
test_data = mx.gluon.data.DataLoader(mx.gluon.data.vision.MNIST(train=False, transform=transform),
```

as

```
def transform(data, label):
    return data.astype(np.float32)/255, label.astype(np.float32)
train_data = mx.gluon.data.DataLoader([d for d in mx.gluon.data.vision.MNIST(train=True, transform=transform)],
                                      batch_size, shuffle=True)
test_data  = mx.gluon.data.DataLoader([d for d in mx.gluon.data.vision.MNIST(train=False, transform=transform)],
                                     batch_size, shuffle=False)
```

but, in general, it would be better to add a 'cache=false' option to the _DownloadedDataset to enable the caching of underlying data.

**Comment by mbaijal** [comment]
Can you please add the labels "Data-loading" and "Feature request" to this issue
